### PR TITLE
Kotlin Cleanup for HelpDialog.kt file

### PR DESCRIPTION
**Pull request by Umesh-01**
## Purpose / Description
Kotlin Cleanup for HelpDialog.kt

## Fixes
Fixes part of #10489 

## Approach
1. Changed val to lateinit var
2. Changed to Kotlin Functional Interface

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt [hunk 1]
```diff
@@ -44,6 +44,7 @@ import androidx.fragment.app.Fragment
 import androidx.fragment.app.FragmentManager
 import androidx.preference.*
 import com.afollestad.materialdialogs.MaterialDialog
+import com.arcao.slf4j.timber.BuildConfig
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt:47]: I don't understand why this import was added? There are no other changes in the file, so an import addition shouldn't be required unless I am missing something?

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/HelpDialog.kt [hunk 2]
```diff
@@ -203,19 +203,21 @@ object HelpDialog {
...
-        private val mFunc: ActivityConsumer?
 
-        constructor(@StringRes titleRes: Int, @DrawableRes iconRes: Int, analyticsRes: String?, func: ActivityConsumer?) : super(titleRes, iconRes, analyticsRes) {
-            mFunc = func
+        private lateinit var mFunc: ActivityConsumer
+
+        constructor(@StringRes titleRes: Int, @DrawableRes iconRes: Int, analyticsRes: String?, func: ActivityConsumer) : super(titleRes, iconRes, analyticsRes) {
+            if (func != null) {
+                mFunc = func
+            }
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/dialogs/HelpDialog.kt:212]: func cannot be null can it? I don't see `ActivityConsumer?` I see `ActivityConsumer` so it appears not-null, and then you don't have to use the `!!` later I think

**Review by mikehardy — CHANGES_REQUESTED** [review]
Thanks for posting this! I left a couple specific questions

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt (hunk 1 above)
Comment (Umesh-01) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt:47]: It was showing some unexpected [errors](https://github.com/ankidroid/Anki-Android/issues/10489#issuecomment-1091981347) when I opened the project. After importing this everything get resolved.

File: AnkiDroid/src/main/java/com/ichi2/anki/dialogs/HelpDialog.kt (hunk 2 above)
Comment (Umesh-01) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/HelpDialog.kt:212]: Yes, I also thought the same. But it was showing error over there and suggested to surround it with if else or use `!!` operator.

Comment (Umesh-01) [reply AnkiDroid/src/main/java/com/ichi2/anki/dialogs/HelpDialog.kt]: I have checked it again, now it's working. Maybe it was my mistake. And I have done the changes.

**Comment by Umesh-01** [comment]
@mikehardy It is showing that `lateinit` is not necessary.

image

when I build project it is showing me this error:

image


And I used `gradlew assemble --stacktrace` command to check what is causing the problem, it was showing me these errors:

image


Now, when I removed the `lateinit` keyword then the BUILD is successful. And output looks like this:

image

Should I commit the changes in this PR??

File: AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt (hunk 1 above)
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/Preferences.kt:47]: This is incorrect

**Comment by david-allison** [comment]
Yes, commit them, otherwise we can't review them

Does it need to be a var? Can we make it a val?

**Comment by Umesh-01** [comment]
> Yes, commit them, otherwise we can't review them
> 
> Does it need to be a var? Can we make it a val?

Yes, we can make it, should I change it to val??

**Comment by david-allison** [comment]
Most of the Kotlin cleanup boils down to: "use your best judgement", `val` sounds good here

**Review by david-allison — APPROVED** [review]
Have you tested this?

If it works, looks great!

**Comment by Umesh-01** [comment]
> Have you tested this?
> 
> If it works, looks great!

Yes it was working.

**Comment by mikehardy** [comment]
I want to add something else though - those images are basically illegible to me on a mobile device. If there is a textual error, it's always better to just paste the text in, inside either triple-backticks enclosure or a `>` for quoting like

```
this is some error message I got and it's huge
so it is in a big block
of quoted text
etc
```

or 

> this small line of text was shown

**Comment by Umesh-01** [comment]
Okk, got it.

**Review by ShridharGoel — APPROVED** [review]

**Review by mikehardy — APPROVED** [review]
Nice!

**Comment by mikehardy** [comment]
Hi there! APRIL OpenCollective Notice

Just a friendly notice that we try to process OpenCollective payments monthly - it's time for [April 2022 submissions](https://github.com/ankidroid/Anki-Android/pulls?q=is%3Apr+merged%3A2022-04-01..2022-04-30)

If you are interested in compensation for this work, the process with details is here:

https://github.com/ankidroid/Anki-Android/wiki/OpenCollective-Payment-Process#how-to-get-paid

I only post one comment per person to avoid spamming you, regardless of the number of PRs merged, but this note applies to all PRs merged for in the month of March.

Please note that GSoC contributions are okay for this process. Our philosophy is that our users have donated to AnkiDroid for all contributions. The only PRs that will not go through the OpenCollective process are ones directly related to am accepted GSoC project from a selected participant, since those receive a stipend from GSoC itself.

Please understand that this means the monthly budget we have for payments might be stretched a bit and the cap on payments-per-person may be lower, but we try to make our process as fair and transparent as possible, we just need your understanding.

Thanks!

**Pull request merged**